ext2/ext2.c

```diff
@@ -5324,6 +5329,62 @@ static int _ext2_fchown(
     return ret;
 }
 
+static int _ext2_lchown(
+    myst_fs_t* fs,
+    const char* pathname,
+    uid_t owner,
+    gid_t group)
+{
+    int ret = 0;
+    ext2_t* ext2 = (ext2_t*)fs;
+    struct locals
+    {
+        ext2_ino_t ino;
+        ext2_inode_t inode;
+        char suffix[PATH_MAX];
+    }* locals = NULL;
+    myst_fs_t* tfs = NULL;
+
+    if (!_ext2_valid(ext2) || !pathname)
+        ERAISE(-EINVAL);
+
+    if (!(locals = malloc(sizeof(struct locals))))
+        ERAISE(-ENOMEM);
+
+    /* Check if path exists */
+    ECHECK(_path_to_inode(
+        ext2,
+        pathname,
+        NOFOLLOW,
+        NULL,
+        &locals->ino,
+        NULL,
+        &locals->inode,
+        locals->suffix,
+        &tfs));
+    if (tfs)
+    {
+        /* delegate operation to target filesystem */
+        ECHECK((ret = tfs->fs_lchown(tfs, locals->suffix, owner, group)));
+        goto done;
+    }
+
+    if (!S_ISLNK(locals->inode.i_mode))
+        ERAISE(-ENOTDIR);
```

Comment:
Does this mean we return -ENOTDIR when the file is not a soft link? I think in that case, we just treat it as a regular file.

Reply:
Thanks for pointing this out. Fixed.